runtime/ieee_754.js

```diff
@@ -284,20 +293,82 @@ function caml_hypot_float (x, y) {
 
 // FIX: these five functions only give approximate results.
 //Provides: caml_log10_float const
-function caml_log10_float (x) { return Math.LOG10E * Math.log(x); }
+var caml_log10_float = Math.log10 || function (x) { return Math.LOG10E * Math.log(x); }
 //Provides: caml_cosh_float const
-function caml_cosh_float (x) { return (Math.exp(x) + Math.exp(-x)) / 2; }
+var caml_cosh_float = Math.cosh || function (x) { return (Math.exp(x) + Math.exp(-x)) / 2; }
+//Provides: caml_acosh_float const
+var caml_acosh_float = Math.acosh || function (x) { return Math.log(x + Math.sqrt(x * x - 1)); }
 //Provides: caml_sinh_float const
-function caml_sinh_float (x) { return (Math.exp(x) - Math.exp(-x)) / 2; }
+var caml_sinh_float = Math.sinh || function (x) { return (Math.exp(x) - Math.exp(-x)) / 2; }
+//Provides: caml_asinh_float const
+var caml_asinh_float = Math.asinh || function (x) { 
+  // Polyfill from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/asinh
+  var absX = Math.abs(x), w
+  if (absX < 3.725290298461914e-9) // |x| < 2^-28
+      return x
+  if (absX > 268435456) // |x| > 2^28
+      w = Math.log(absX) + Math.LN2
+  else if (absX > 2) // 2^28 >= |x| > 2
+      w = Math.log(2 * absX + 1 / (Math.sqrt(x * x + 1) + absX))
+  else
+      var t = x * x, w = Math.log1p(absX + t / (1 + Math.sqrt(1 + t)))
+
+  return x > 0 ? w : -w
+}
+
 //Provides: caml_tanh_float const
-function caml_tanh_float (x) {
+var caml_tanh_float = Math.tanh || function (x) {
   var y = Math.exp(x), z = Math.exp(-x);
   return (y - z) / (y + z);
 }
 
-//Provides: caml_round_float
+//Provides: caml_atanh_float const
+var caml_atanh_float = Math.atanh || function (x) {
+  return Math.log((1+x)/(1-x)) / 2;
+}
+
+
+//Provides: caml_round_float const
 function caml_round_float (x) { return Math.round(x); }
 
+//Provides: caml_cbrt_float const
+var caml_cbrt_float = Math.cbrt || (function (pow) { 
+  return function cbrt(x) {
+    return x < 0 ? -pow(x, 1/3) : pow(x, 1/3);
+  };
+})(Math.pow);
```

Comment:
Why pass pow in like this?

Reply:
All these polyfill implementations are taken from the MDN docs. This one, in particular was taken from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/cbrt